packages/antd/src/FileUploadAnt.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import Dropzone from 'react-dropzone';
+import { Spin } from 'antd';
+import { InboxOutlined } from '@ant-design/icons';
+
+import './FileUploadStyles.css';
+import type { FileUpload } from '@moxb/moxb';
+import { LabelAnt } from './LabelAnt';
+import { ActionButtonAnt } from './ActionAnt';
+
+export interface FileUploadProps {
+    operation: FileUpload;
+}
+
+export class FileUploadAnt extends React.Component<FileUploadProps> {
+    componentDidMount() {
+        this.props.operation.reset();
+    }
+
+    onDrop(files: File[]) {
+        if (files.length) {
+            this.props.operation.upload(files[0]);
+        }
+    }
+
+    constructor(props: FileUploadProps) {
+        super(props);
+        this.onDrop = this.onDrop.bind(this);
+    }
+
+    render() {
+        const {
+            succeeded,
+            successLabel,
+            errorLabel,
+            progressLabel,
+            resetAction,
+            pending,
+            prompt,
+            visible,
+        } = this.props.operation;
+
+        return (
+            visible && (
+                <div>
+                    <LabelAnt operation={progressLabel} />
+                    {pending && <Spin />}
+                    <LabelAnt operation={errorLabel} className="file-upload-fail" />
+                    <LabelAnt operation={successLabel} className="file-upload-ok" />
+                    {!(succeeded || pending) && (
+                        <Dropzone onDrop={this.onDrop} maxFiles={1}>
+                            {({ getRootProps, getInputProps }) => (
+                                <section>
+                                    <div {...getRootProps()}>
+                                        <input {...getInputProps()} />
+                                        {/*Using classes from https://ant.design/components/upload/ */}
```

Comment:
why not using the https://ant.design/components/upload/ directly?

Reply:
No special reason. We could probably do that, too. I just did [a search on npmjs fo  upload](https://www.npmjs.com/search?q=upload), and selected the second most popular solution, because it seemed to be compatible with our technologies.

We could probably re-implement it using the native antd tools, but I don't feel like spending any time on this, since what we have is already working.